src/platform.ts

```diff
@@ -66,20 +66,49 @@ export function getDefaultPowerShellPath(
     use32Bit: boolean = false): string | null {
 
     let powerShellExePath;
+    let psCoreInstallPath;
 
-    // Find the path to powershell.exe based on the current platform
+     // Find the path to the powershell executable based on the current platform
     // and the user's desire to run the x86 version of PowerShell
     if (platformDetails.operatingSystem === OperatingSystem.Windows) {
         if (use32Bit) {
-            powerShellExePath =
-                platformDetails.isOS64Bit && platformDetails.isProcess64Bit
-                    ? SysWow64PowerShellPath
-                    : System32PowerShellPath;
+            psCoreInstallPath =
+                (platformDetails.isProcess64Bit ? process.env["ProgramFiles(x86)"] : process.env.ProgramFiles)
+            + "\\PowerShell";
         } else {
-            powerShellExePath =
-                !platformDetails.isOS64Bit || platformDetails.isProcess64Bit
-                    ? System32PowerShellPath
-                    : SysnativePowerShellPath;
+            psCoreInstallPath =
+                (platformDetails.isProcess64Bit ? process.env.ProgramFiles : process.env.ProgramW6432) + "\\PowerShell";
+        }
+
+        if (fs.existsSync(psCoreInstallPath)) {
+            const arch = platformDetails.isProcess64Bit ? "(x64)" : "(x86)";
+            const psCorePaths =
+                fs.readdirSync(psCoreInstallPath)
+                .map((item) => path.join(psCoreInstallPath, item))
+                .filter((item) => {
+                    const exePath = path.join(item, "pwsh.exe");
+                    return fs.lstatSync(item).isDirectory() && fs.existsSync(exePath);
+                })
+                .map((item) => ({
+                    versionName: `PowerShell Core ${path.parse(item).base} ${arch}`,
+                    exePath: path.join(item, "pwsh.exe"),
+                }));
+
+            if (psCorePaths) {
+                return powerShellExePath = psCorePaths[0].exePath;
+            }
+        }
+
+        if (use32Bit) {
+                powerShellExePath =
+                    platformDetails.isOS64Bit && platformDetails.isProcess64Bit
+                        ? SysWow64PowerShellPath
+                        : System32PowerShellPath;
+        } else {
+                powerShellExePath =
+                    !platformDetails.isOS64Bit || platformDetails.isProcess64Bit
```

Comment:
instead of storing this to a variable, you can return directly from here:

```suggestion
                return !platformDetails.isOS64Bit || platformDetails.isProcess64Bit
```